src/Future/functions.php

```diff
@@ -81,7 +124,7 @@ function some(iterable $futures, int $count, ?Cancellation $cancellation = null)
         }
     }
 
-    if (empty($errors)) {
+    if (\count($values) + \count($errors) < $count) {
```

Comment:
I'm not sure whether this should really be an `\Error`, IMO this should probably be an exception, as the consumer might not know how many elements are in the iterator.

Reply:
I agree - an Exception is better.

Reply:
Which exception should we throw here? `CompositeException` currently requires a non-empty list of exceptions, but I'm not sure whether that's a good idea, as we could use it here without having any errors if there aren't any futures in the iterable.

Reply:
Seems fine to allow an empty list.